include/dr/shp/algorithms/for_each.hpp

```diff
@@ -13,84 +13,34 @@
 
 namespace shp {
 
-template <typename ExecutionPolicy, lib::distributed_contiguous_range R,
-          typename Fn>
-void for_each(ExecutionPolicy &&policy, R &&r, Fn &&fn) {
+template <lib::distributed_contiguous_range R, typename Fn>
```

Comment:
I understand the urge to simplify, but I think we actually do want to leave the signature as `typename ExecutionPolicy`, as we will be adding other execution policies later, and we also want it to resemble the [standard library's interface](https://en.cppreference.com/w/cpp/algorithm/for_each).

Reply:
Restored `ExecutionPolicy` template in order to give nicer compilation errors.
Removed unnecessary `if constexpr` as we discussed in the call.